realm/realm-library/src/main/java/io/realm/OrderedRealmCollection.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import java.util.List;
+
+/**
+ * A {@code OrderedRealmCollection} is a collection which maintains an ordering for its elements. Every
+ * element in the {@code OrderedRealmCollection} has an index. Each element can thus be accessed by its
+ * index, with the first index being zero. Normally, {@code OrderedRealmCollection}s allow duplicate
+ * elements, as compared to Sets, where elements have to be unique.
+ */
+public interface OrderedRealmCollection<E extends RealmObject> extends List<E>, RealmCollection<E> {
+
+    /**
+     * Gets the first object from the collection.
+     *
+     * @return the first object or {@code null} if the collection is empty.
+     * @throws IndexOutOfBoundsException if the collection is empty.
+     */
+    E first();
+
+    /**
+     * Gets the last object from the collection.
+     *
+     * @return the last object or {@code null} if the collection is empty.
+     * @throws IndexOutOfBoundsException if the collection is empty.
+     */
+    E last();
+
+    /**
+     * Sorts (ascending) a collection..
```

Comment:
An extra `.` at the end. Also can make it more formal like `Sorts a collection in ascending order.`